verification/refinement.md

```diff
@@ -1,2 +1,115 @@
+{
+  "state mapping": {
+    "ILA_state_1" : "Verilog_state_1",                                  // case 1
+    "ILA_state_2" : "ILA_state_2 == Verilog_state_2 + Verilog_state_3", // case 2
+    "ILA_state_3" : ["__START__", "Verilog_state_3"],                   // case 3.a
+    "ILA_state_4" : { "cond":"__START__", "map":"Verilog_state_4"},     // case 3.b
+    "ILA_state_5" : [["__START__", "Verilog_state_5"], ["__IEND__","Verilog_state_6"]],     // case 4
+    // more ...
+    }, 
+    // other fields ...
+}
+```
+
+One note in the above example: the condition can refer to special signals (`__START__ ` and `__IEND__`), which are the condition when the instruction-under-verification starts to execute and finishes.
+
+### Instruction Completion Condition ###
+
+
+The instruction completion condition is specified per instruction. In the JSON file, it is a list of maps. The list does not need to be a full list of instructions. Those not in the list will not be verified. Each dictionary must have an element whose key is `instruction` and the value of this element is the name of the instruction in the ILA model. Besides this name element, it must contain one of the `ready bound` or the `ready signal` element. The `ready bound`  specifies a bound that the instruction takes. It is used for instructions that take a fixed number of cycles. Alternatively, one can provide a signal (or a predicate) in the `ready signal` field.
+
+There are several optional fields. They are `start condition`, `max bound`, `flush constraint`, `pre flush-end` and `post flush-end`. 
+
+The `start condition` field, if provided, should be a list of strings. Each string is a Verilog expression that acts as a predicate on the Verilog design. It can be used to constrain the Verilog implementation state, because usually there are more microarchitectural (implementation states) in the design. For the starting state of an instruction, these microarchitectural states should be `consistent` with the visible states in the ILA, and you can use this field to enforce the `consistency` at your discretion. You can use `$decode$` and `$valid$` to refer to the instruction's decode function or its ILA valid condition.
+
+The `max bound` can be used when `ready signal` field is provided. It provides an assumption that the condition that the `ready signal` field specified will occur in the given number of cycles, and this will limit the model checking to that bound. Usually this is used for the additional environment assumptions about the input (for example, how many cycles at most there are for a request to be served with a response).
+
+The `flush constraint`, `pre flush-end` and `post flush-end` signals are used when using the `flushing verification setting`. For this verification setting, you can refer to \href{https://arxiv.org/abs/1801.01114}{our paper} on the ILA-based verification or the \href{https://dl.acm.org/citation.cfm?id=735662}{Burch-Dill's approach} on processor verification.
```

Comment:
Latex-styled reference/hyperlink are not supported in markdown. 
There's a copy of the final official published TODAES paper in this repo to be used/referenced.